src/vision/src/CircleDetection.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+ * Created By: Robyn Castro
+ * Created On: June 17, 2017
+ * Description: Determines whether or not a circle is seen on
+ *              the screen then publishes a recommended twist
+ *              message.
+ */
+
+#include <CircleDetection.h>
+#include <image_transport/image_transport.h>
+#include <std_msgs/Bool.h>
+
+using namespace cv;
+using namespace std;
+using namespace cv_bridge;
+
+CircleDetection::CircleDetection(std::string &image_path) {
+
+    cv::Mat bgr_image = imread(image_path);
+
+    // Check if the image can be loaded
+    check_if_image_exist(bgr_image, image_path);
+
+    minTargetRadius = 20;
+    countCircles(bgr_image);
+
+    waitKey(0);
+
+}
+
+CircleDetection::CircleDetection() {
+    minTargetRadius = 20;
+}
+
+CircleDetection::CircleDetection(int argc, char **argv, std::string node_name) {
+
+    // Setup handles
+    ros::init(argc, argv, node_name);
+    ros::NodeHandle nh;
+    ros::NodeHandle private_nh("~");
+
+    // Setup image transport
+    image_transport::ImageTransport it(nh);
+
+    // Setup subscriber
+    std::string image_topic = "/robot/vision/filtered_image";
+    int refresh_rate = 10;
+    image_sub = it.subscribe<CircleDetection>(image_topic, refresh_rate,
+                                              &CircleDetection::filteredImageCallBack, this);
+
+    // Setup publishers
+    std::string output_topic = "/robot/vision/activity_detected";
+    uint32_t queue_size = 1;
+    activity_publisher = private_nh.advertise<std_msgs::Bool>(output_topic, queue_size);
```

Comment:
I think I mentioned something similar before about using `private_nh` but having `output_topic` itself being global.
Having a private nodehandle here won't do anything since the output topic is prefixed with a `/`